quarkchain/cluster/shard_state.py

```diff
@@ -774,6 +787,7 @@ def run_block(self, block, evm_state=None, x_shard_receive_tx_list=None):
         for k, v in pure_coinbase_amount.balance_map.items():
             evm_state.delta_token_balance(evm_state.block_coinbase, k, v)
 
+        evm_state.add_block_header(block.header)
```

Comment:
Why do we need to add a block header here?

Reply:
after running the block, evm_state will commit the underlying changes, while we also want to make sure its `prev_header` is up to date, so the next time evm_state executing txs, it will have the latest block header available